### PR TITLE
BREAKING CHANGE: Fixes and updates configuration to work

### PR DESCRIPTION
**Pull request by dploeger**
with current Azure provider

### CODE REVIEW

**Review by rexhinvorpsi — APPROVED** [review]

File: main.tf [hunk 1]
```diff
@@ -10,22 +10,17 @@ resource "azurerm_mssql_server" "azuresqldbsrv" {
...
 
 resource "azurerm_mssql_database" "azuresqldb" {
-  name                             = "${lower(var.project)}${lower(var.stage)}db${var.suffix}"
-  server_id                        = azurerm_mssql_server.azuresqldbsrv.id
-  location                         = var.location
-  resource_group_name              = var.resource_group
-  server_name                      = azurerm_mssql_server.azuresqldbsrv.name
-  edition                          = var.edition
-  requested_service_objective_name = var.performance_class
+  name      = "${lower(var.project)}${lower(var.stage)}db${var.suffix}"
```
Comment (ThomasMinor) [inline main.tf]: is it possible to pass the whole string to the lower function here?
`name = lower("${var.project}${var.stage}db${var.suffix}")`

File: main.tf [hunk 2]
```diff
@@ -10,22 +10,17 @@ resource "azurerm_mssql_server" "azuresqldbsrv" {
...
 
 resource "azurerm_mssql_firewall_rule" "azure-sql-enable-access-firewall" {
-  name                = "${lower(var.project)}${lower(var.stage)}dbsrvaccessfromazure${var.suffix}"
-  server_id           = azurerm_mssql_server.azuresqldbsrv.id
-  resource_group_name = var.resource_group
-  server_name         = azurerm_mssql_server.azuresqldbsrv.name
-  start_ip_address    = "0.0.0.0"
-  end_ip_address      = "0.0.0.0"
+  count            = var.public_access ? 1 : 0
+  name             = "${lower(var.project)}${lower(var.stage)}dbsrvaccessfromazure${var.suffix}"
```
Comment (ThomasMinor) [inline main.tf]: is it possible to pass the whole string to the lower function here?

`name = lower("${var.project}${var.stage}dbsrvaccessfromazure${var.suffix}")`

**Review by ThomasMinor — CHANGES_REQUESTED** [review]

**Review by timdeluxe — APPROVED** [review]

File: main.tf (hunk 2 above)
Comment (dploeger) [reply main.tf]: done

File: main.tf (hunk 1 above)
Comment (dploeger) [reply main.tf]: done

**Review by ThomasMinor — APPROVED** [review]

**Pull request merged**